packages/sanity/src/core/form/inputs/files/FileInput/FileActionsMenu.tsx

```diff
@@ -101,16 +116,17 @@ export function FileActionsMenu(props: Props) {
 
       <Box padding={2}>
         <Flex justify="center">
+          {/* Using a customized Popover instead of MenuButton because a MenuButton will close on click
+     and break replacing an uploaded file. */}
           <Popover
-            content={
-              <Menu ref={setMenuElement} shouldFocus="first">
-                {children}
-              </Menu>
-            }
+            content={<Menu ref={setMenuElement}>{children}</Menu>}
+            id="file-actions-menu"
             portal
             open={isMenuOpen}
+            constrainSize
```

Comment:
**note:** another missed fix from the image actions menu – perhaps we should consider consolidating both of these action menus in future